sdk/servicebus/azure-servicebus/azure/servicebus/aio/management/_management_client_async.py

```diff
@@ -0,0 +1,853 @@
+        :keyword timedelta lock_duration: The value you want to update to.
+        :keyword bool dead_lettering_on_message_expiration: The value you want to update to.
+        :keyword timedelta duplicate_detection_history_time_window: The value you want to update to.
+        :keyword int max_delivery_count: The value you want to update to.
+        :rtype: ~azure.servicebus.management.QueueDescription
+        """
+
+        if not isinstance(queue_description, QueueDescription):
+            raise TypeError("queue_description must be of type QueueDescription")
+
+        internal_description = queue_description._to_internal_entity()
+        to_update = copy(internal_description)
+
+        to_update.default_message_time_to_live = kwargs.get(
+            "default_message_time_to_live") or queue_description.default_message_time_to_live
+        to_update.lock_duration = kwargs.get("lock_duration") or queue_description.lock_duration
+        to_update.dead_lettering_on_message_expiration = kwargs.get(
+            "dead_lettering_on_message_expiration") or queue_description.dead_lettering_on_message_expiration
+        to_update.duplicate_detection_history_time_window = kwargs.get(
+            "duplicate_detection_history_time_window") or queue_description.duplicate_detection_history_time_window
+        to_update.max_delivery_count = kwargs.get("max_delivery_count") or queue_description.max_delivery_count
+
+        to_update.default_message_time_to_live = avoid_timedelta_overflow(to_update.default_message_time_to_live)
+        to_update.auto_delete_on_idle = avoid_timedelta_overflow(to_update.auto_delete_on_idle)
+
+        create_entity_body = CreateQueueBody(
+            content=CreateQueueBodyContent(
+                queue_description=to_update,
+            )
+        )
+        request_body = create_entity_body.serialize(is_xml=True)
+        with _handle_response_error():
+            await self._impl.entity.put(
+                queue_description.name,  # type: ignore
+                request_body,
+                api_version=constants.API_VERSION,
+                if_match="*",
+                **kwargs
+            )
+
+    async def delete_queue(self, queue: Union[str, QueueDescription], **kwargs) -> None:
+        """Delete a queue.
+
+        :param Union[str, azure.servicebus.management.QueueDescription] queue: The name of the queue or
+         a `QueueDescription` with name.
+        :rtype: None
+        """
+        try:
+            queue_name = queue.name  # type: ignore
+        except AttributeError:
+            queue_name = queue
+        if not queue_name:
+            raise ValueError("queue_name must not be None or empty")
+        with _handle_response_error():
+            await self._impl.entity.delete(queue_name, api_version=constants.API_VERSION, **kwargs)
+
+    def list_queues(self, **kwargs) -> AsyncItemPaged[QueueDescription]:
```

Comment:
Naive question #<I'velosttrack>

Do ItemPaged behave similarly enough to a native collection that this will be consumable without further detail in the docstring?  If other SDKs return paged and don't specify outside of the typedef I'll take that as a vote of confidence, I'm just reading this function and thinking that a user will go "for queue in list_queues" and be surprised when it doesn't 'go'.

Reply:
If a user writes `for queue in list_queues:` it will work. And this is how we've documented it in our other SDKs.
(However the :rtype: tag needs updating for the Async object)

Reply:
I added returns in docstring by reference storage.
:returns: An iterable (auto-paging) response of QueueDescription.